proposals/protected-components.txt

```diff
@@ -0,0 +1,245 @@
+     However, subgroup did not like the alternative of disallowing
+     protected components in extensions of parent types without any
+     protected component.
+
+L3. No intrinsic assignment to an object whose declared type is protected,
+    or that has an ultimate component of protected type, outside the module
+    in which the protected component is defined. (cf. 19-135r1)
+
+M2. A function defined outside the module may have a result variable of a
+    protected type.
+    - This is same as ordinary local variables (case A).
+    - The function result shall not be a pointer.
+    - The function result will (hopefully) be finalised after its use.
+
+N2. Structure constructor outside the module in which the type is allowed.
+
+O. Another type may have a component of protected type, and thereby
+   inherits the protected type attribute.
+   - The protection rules apply to types with "protected potential
+     subobject components" not just types with immediate protected
+     components.
+
+P2. Extension of a protected type is permitted.
+
+Q. An extension type may have a protected attribute even if the parent type
+   does not have a protected attribute. Requiring the parent type to have
+   be protected is too restricted for many uses.
+
+S. An object with a protected declared type is prohibited to be the target
+   of an unlimited polymorphic pointer.
+   - This rule applies even inside the module, even if the polymorphic
+     pointer is private, because its target might thereafter become
+     associated with a public polymorphic pointer.
+
+T. If an actual argument with a protected declared type is associated with
+   an unlimited polymorphic dummy, the dummy shall have INTENT(IN) or
+   INTENT(INOUT) and shall not have TARGET (or POINTER, which implies
+   TARGET).
+   - If the dummy has TARGET it might become associated with an unlimited
+     polymorphic pointer.
+   - If the dummy has POINTER, its target might eventually become
+     associated with an unlimited polymorphic pointer (see P.)
+
+
+3. Syntax
+=========
+
+3.1 Protected Components
+
+One may add the PROTECTED attribute to component definitions.
+
+    type :: foo
+      real, protected :: a
+      real, private :: b
+      real, public :: c
+    end type foo
+
+The requirement R allows a type-wide PROTECTED status, which may be
+overridden for an individual component using another access specifier.
+
+    type :: foo
+      protected
+      real, private :: a
+      real :: b
+    end type foo
+
+
+3.2 Protected Types
+
+One may add the PROTECTED attribute to type definitions. One may
+independently control the access specifier of each component.
+
+    type, protected :: foo
+      real, public :: a
+      real, private :: b
+      real, protected :: c
+    end type foo
+
+The PROTECTED keyword meaning different things in these contexts may be
+confusing. It's worth considering different keywords (e.g., readonly
+components).
```

Comment:
or visible components

Reply:
Thanks, I'll add that as another option. I think making clear that we're very open to other/more appropriate keywords will make it stronger.